src/systems/systems/src/pressurization/cabin_pressure_controller.rs

```diff
@@ -0,0 +1,475 @@
+    length::{foot, meter},
+    pressure::{hectopascal, inch_of_mercury, psi},
+    ratio::percent,
+    velocity::{foot_per_minute, knot, meter_per_second},
+};
+
+pub struct CabinPressureController {
+    pressure_schedule_manager: PressureScheduleManager,
+    outflow_valve_open_amount: Ratio,
+    exterior_pressure: Pressure,
+    cabin_pressure: Pressure,
+    last_cabin_pressure: Pressure,
+    departure_elev: Length,
+    landing_elev: Length,
+    is_ldg_elev_auto: bool,
+    cabin_alt: Length,
+    aircraft_vs: Velocity,
+    cabin_target_vs: Velocity,
+    cabin_vs: Velocity,
+    is_standby: bool,
+    // is_faulted: bool,
+}
+
+impl CabinPressureController {
+    pub fn new() -> Self {
+        Self {
+            pressure_schedule_manager: PressureScheduleManager::new(),
+            outflow_valve_open_amount: Ratio::new::<percent>(100.),
+            exterior_pressure: Pressure::new::<inch_of_mercury>(29.92),
+            cabin_pressure: Pressure::new::<inch_of_mercury>(29.92),
+            cabin_alt: Length::new::<meter>(0.),
+            aircraft_vs: Velocity::new::<meter_per_second>(0.),
+            cabin_target_vs: Velocity::new::<meter_per_second>(0.),
+            cabin_vs: Velocity::new::<meter_per_second>(0.),
+            last_cabin_pressure: Pressure::new::<inch_of_mercury>(29.92),
+            departure_elev: Length::new::<foot>(-5000.),
+            landing_elev: Length::new::<meter>(0.),
+            is_ldg_elev_auto: true,
+            is_standby: true,
+        }
+    }
+
+    pub fn update(
+        &mut self,
+        context: &UpdateContext,
+        outflow_valve: &PressureValve,
+        overhead: &PressurizationOverheadPanel,
+        aircraft_inputs: &AircraftInputsPressurization,
+    ) {
+        self.pressure_schedule_manager.update(
+            context,
+            aircraft_inputs.eng_1_n1(),
+            aircraft_inputs.eng_2_n1(),
+        );
+        self.outflow_valve_open_amount = outflow_valve.open_amount();
+        self.exterior_pressure = context.ambient_pressure();
+        self.aircraft_vs = context.vertical_speed();
+        self.last_cabin_pressure = self.cabin_pressure;
+        self.update_cabin_pressure(context);
+        self.update_cabin_altitude(
+            aircraft_inputs.sea_level_pressure(),
+            aircraft_inputs.destination_qnh(),
+        );
+        self.update_cabin_vs(context); //Pre-smooth function
+        self.set_cabin_vs(context); //Smooth function
+        self.update_departure_elev(context);
+        self.update_landing_elev(aircraft_inputs.landing_elev());
+        self.is_ldg_elev_auto = !overhead.mode_is_man();
+        self.update_active_system(context);
+    }
+
+    fn update_cabin_pressure(&mut self, context: &UpdateContext) {
+        // Convert cabin V/S to pressure/delta
+        const KPA_FT: f64 = 0.0366; //ASL
+
+        let cab_press_var =
+            -self.cabin_vs.get::<foot_per_minute>() * KPA_FT / 60. * context.delta().as_secs_f64();
```

Comment:
**suggestion**: Use `delta_as_secs_f64()` instead of `delta().as_secs_f64()`. This same suggestion is applicable in various other usages.

```suggestion
            -self.cabin_vs.get::<foot_per_minute>() * KPA_FT / 60. * context.delta_as_secs_f64();
```

Reply:
The order of mathematical operations is correct. The `60.` in this formula represented seconds (i.e. 60 seconds in a minute, to convert from feet/minute to feed/second). I've now changed this to directly get the vertical speed in `feet_per_second` through the uom module so the `60.` can be removed.

Reply:
All `delta().as_secs_f64()` swapped for `delta_as_secs_f64()`.